core/modules/system/system.admin.inc

```diff
@@ -2328,6 +2328,244 @@ function system_status($check = FALSE) {
+  );
+
+  $debug_info = system_get_debug_info();
+
+  $form['debug_info'] = array(
+    '#type' => 'textarea',
+    '#title' => t('Debug report'),
+    '#value' => _system_format_array_to_plain_text_columns($debug_info, array('rtl' => $language->direction == LANGUAGE_RTL)),
+    // Expand the default length of the text field, to ensure that all the
+    // debug output is visible without having to scroll within the text field.
+    '#rows' => count($debug_info) + 1,
+    '#attributes' => array(
+      'id' => 'debug-info-wrapper',
+      // Using the 'readonly' attribute to lock the text, since it gives us a
+      // "cursor: text" cursor for free (using 'disabled' would result in a
+      // "cursor: text" instead).
+      'readonly' => 'readonly',
+    ),
+  );
+
+  return $form;
+}
+
+/**
+ * Compiles the various information for the debug info page.
+ *
+ * @see system_debug_info()
+ *
+ * @since 1.28.0 function added.
+ */
+function system_get_debug_info() {
+  global $language;
+
+  // An array that is later passed to sprintf() so that the final output can be
+  // formatted in two columns of plain text. The array keys will be the first
+  // column and the values will be the second column. The number of spaces
+  // between the two columns is dynamically calculated, based on the lenght of
+  // the string of the first column. This ensures that the strings in the second
+  // column all start at the same point and appear as vertically aligned.
+  $debug_info_lines = array();
+
+  $empty_line = NULL;
+
+  $jquery = backdrop_get_library('system', 'jquery');
+  $jquery_ui = backdrop_get_library('system', 'ui');
+
+  // Main system info.
+  $system_info = array(
+    'Backdrop CMS:' => BACKDROP_VERSION,
+    t('Installation profile:') => backdrop_get_profile(),
+    t('PHP version:') => phpversion(),
+    t('Database server:') => Database::getConnection()->version(),
+    t('Web server:') => check_plain($_SERVER['SERVER_SOFTWARE']),
+    t('jQuery version:') => $jquery['version'],
+    t('jQuery UI version:') => $jquery_ui['version'],
+  );
+
+  foreach ($system_info as $info => $value) {
+    $debug_info_lines[$info] = $value;
+  }
+
+  $debug_info_lines[] = $empty_line;
+
+  // Text editors info.
+  $text_editor_usage = array();
+  $text_editors = filter_get_editors();
+  $text_formats = filter_formats(NULL, TRUE);
+  // Get the current list of all modules available on the site.
+  $modules = system_rebuild_module_data();
+
+  foreach ($text_formats as $format => $format_info) {
+    if (!empty($format_info->editor)) {
+      $editor = $text_editors[$format_info->editor];
+      $editor_label = $editor['label'];
+      // Add a suffix with the editor version or the editor library version
+      // to the editor label.
+      if (!empty($editor['library_version'])) {
+        $editor_label .= ', ' . $editor['library_version'];
```

Comment:
I don't think this is exactly right, but something like this would save the need to specify library_version in two places:

```suggestion
      if (!empty($editor['library'])) {
        $library = backdrop_get_library($editor['library']);
        $editor_label .= ', ' . $library['version'];
```

Reply:
This doesn't work, as I noted in previous comments, so resolving this.